Add discussion about parallelization/concurrency and tools integration to vision document draft (#425)
apple/swift-testing: Documentation/Vision.md

isolation.

However, some tests may need to disable parallelization and run one at a time.
It should be possible to opt-out, and this may be especially useful while
migrating from older testing systems which don't support parallelization.
Although opting-out of this behavior should be possible, it should be narrowly
scoped to not sacrifice other tests' ability to run in parallel.

In addition to running tests in parallel relative to each other, tests
themselves should seamlessly support Swift's concurrency features. In particular,
this means:

* Tests should be able to use async/await whenever necessary.
* Tests should support isolation to a global actor such as `@MainActor`, but be
  nonisolated by default. (Isolation by default would undermine the goal of
  running tests in parallel by default.)
* Values passed as arguments to parameterized tests should be `Sendable`, since
  they may cross between isolation domains within the testing system's execution
  machinery.
* Types containing tests functions and their stored properties need not be
  `Sendable`, since they are only used from a single isolation domain while each
  test function is run.

### Tools integration

> A well-rounded testing library should be integrated with popular tools in the

Reviewer: ```suggestion
A well-rounded testing library should be integrated with popular tools used by
```

community. This integration should include some essential functionality such as:

* Building tests into products which can be be executed.
* Running all built tests.
* Showing per-test results, including details of each individual failure during
  a test.
* Showing an aggregate summary of a test run, including failure statistics.

Beyond the essentials, tools may offer other useful features, such as:

* Filtering tests by name, specific traits (e.g. custom tags), or other criteria.

> * Outputting results to a standard format such as [xUnit XML](https://xunit.net/docs/format-xml-v2)

Reviewer: We actually output JUnit-style XML (as does corelibs-xctest.)
